lib/TransferJobs/SubDirectoryTransfer.cs

```diff
@@ -350,11 +350,19 @@ private void CreateDestinationDirectory(CancellationToken cancellationToken)
                 }
                 catch (StorageException storageException)
                 {
-                    throw new TransferException(TransferErrorCode.FailToVadlidateDestination,
-                        string.Format(CultureInfo.CurrentCulture,
-                            Resources.FailedToValidateDestinationException,
-                            storageException.ToErrorDetail()),
-                        storageException);
+                    if ((null != storageException.InnerException)
```

Comment:
Would you like to see if other place will wrapper OperationCanceledException to storageException in cancel?